### PR TITLE
Signup screen

### PR DESCRIPTION
**Pull request by yvanpdev**
users can now create an account.
data will be saved in firebase database with user auth.
added loading animation when sign up button pressed.

### CODE REVIEW

File: src/components/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+checkPassword(text) {
+  this.setState({ password: text });
+
+  if (text.length < 6) {
+    this.setState({
+      passwordLength: 'password must be at least 6 characters',
+      passwordShort: true
+    });
+  } else if (text.length < 25 && text.length > 5) {
+    this.setState({
```
Comment (AryanJ-NYC) [inline src/components/SignUp.js]: What is 5? 6? 25?

As the creator of this code, you know exactly what your intent is. As a reader of this code, I haven't a clue what these numbers represent. These are examples of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)).

In the `src` folder, create a `constants.js`. In that file, export constants such as `MAXIMUM_PASSWORD_LENGTH` and `MINIMUM_PASSWORD_LENGTH`. Then import those constants into this file and use them.

File: src/components/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        <TextInput
+          underlineColorAndroid={'rgba(0,0,0,0.35)'}
+          autoCorrect={false}
+          style={[styles.inputStyle, this.state.passwordShort && styles.shortStyle]}
+          secureTextEntry
+          ref={passwordRef => this.passwordRef = passwordRef}
+          onSubmitEditing={() => this.passwordRef.focus()}
+          returnKeyType='next'
+          placeholder="Password"
+          onChangeText={this.checkPassword.bind(this)}
```
Comment (AryanJ-NYC) [inline src/components/SignUp.js]: No need to `.bind(this)`. It's bound.

File: src/components/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+     passwordShort: true,
+     loading: false
+  }
+
+onButtonPress() {
+  const { email, password, firstName, lastName, userName } = this.state;
+
+  this.setState({ loading: true });
+
+firebase.auth().createUserWithEmailAndPassword(email, password)
```
Comment (AryanJ-NYC) [inline src/components/SignUp.js]: Indent. Are you using a linter?

File: src/components/Login.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+class Login extends Component {
+  state = {
+     email: '',
+     password: '',
+     error: '',
+     loading: false
+  }
+
+  onLogginSuccess() {
+    this.setState({ email: '', password: '', loading: false, error: '' });
```
Comment (AryanJ-NYC) [inline src/components/Login.js]: Typo: loggin, not login

File: src/components/Login.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    Actions.main();
+  }
+
+  onLogginFail() {
+    this.setState({ error: 'Authentication Failed.', loading: false });
```
Comment (AryanJ-NYC) [inline src/components/Login.js]: Typo: loggin, not login

File: src/components/SignUp.js (hunk 2 above)
Comment (cmenedes) [reply src/components/SignUp.js]: In this component we don't have a constructor(perhaps we should?) - so I don't think anything is being bound. If I take away the bind I get an error linked to update being called infinitely. Currently it works as it is written.

**Pull request merged**